### ISSUE TITLE
`gsl-config --prefix` Wrongly Returns "~/.cpanm/work" Temp Directory

### ISSUE THREAD

**Issue by wbraswell** [PerlAlien/Alien-GSL#5]
After I have successfully installed Alien::GSL, I should be able to call `gsl-config --prefix` and have it return the exact same value as Alien::GSL->dist_dir(), however it is incorrectly returning a temporary work directory used by CPANM during build & installation...

CORRECT RETURN VALUE:
```
    $ perl -e 'use Alien::GSL; print Alien::GSL->dist_dir(), "\n";'
    /home/foo_user/perl5/lib/perl5/auto/share/dist/Alien-GSL
```

INCORRECT RETURN VALUE:
```
    $ perl -e 'use Alien::GSL; use Env qw(@PATH); unshift @PATH, Alien::GSL->bin_dir(); print `gsl-config --prefix`;'
    /home/foo_user/.cpanm/work/foo_serial/Alien-GSL-1.01/blib/lib/auto/share/dist/Alien-GSL
```

This causes a potential downstream error in Math::GSL because I am in the process of adding Alien::GSL support but Math::GSL wants to call `gsl-config --prefix`:

https://github.com/leto/math--gsl/blob/master/Build.PL#L321

[[[ UPDATED: system install info ]]]

It is not correct to replace `gsl-config --prefix` with Alien::GSL->dist_dir() because in a system install of GSL the value of bin_dir() is empty and calling dist_dir() actually gives an error:

```
    $ perl -e 'use Alien::GSL; print Alien::GSL->dist_dir(), "\n";'
    Failed to find share dir for dist 'Alien-GSL' at -e line 1.
```

Anyway, we should not be having a patently incorrect value returned by `gsl-config --prefix` because it could cause other unknown future errors in addition to this example.

I do not understand the internals of Alien whatsoever, so I apologize that I do not have a pull request to fix this issue, although I will be more than happy to assist in reaching a solution in any way I can.  Thanks in advance!

:-)

**Comment by plicease** [comment]
Converting this alien over to use `alienfile` and `Alien::Build` may correct the output, as it handles staged builds better than the old `AB::MB` system.

However, I think the work required for this should be taken into consideration.  The prefix is not usually useful in code that is using an alienized library.  For system installs include, lib and other directories can be configured arbitrarily for example in debian libs are in `/usr/lib/$arch`, not `$prefix/lib`.

**Comment by wbraswell** [comment]
Okay I will leave this issue open, for now I will attempt to work around it by disabling any usage of the `gsl-config --prefix` command.

**Comment by plicease** [comment]
This should be addressed by #11
